frb_codegen/src/generator/rust/ty.rs

```diff
@@ -115,7 +114,7 @@ impl<'a> TypeRustGenerator<'a> {
             StructRef(ir) => TypeStructRefGenerator { ir, context }.into(),
             Boxed(ir) => TypeBoxedGenerator { ir, context }.into(),
             EnumRef(ir) => TypeEnumRefGenerator { ir, context }.into(),
-            SyncReturn(ir) => TypeSyncReturnGenerator { ir, context }.into(),
+            SyncReturn(ir) => TypeRustGenerator::new(ir.into_inner(), ir_file, config),
```

Comment:
So here's the thing: I realized I switched patterns here and I removed `TypeSyncReturnGenerator` on purpose in oder to use `TypeRustGenerator` recursively instead.

Keeping `TypeSyncReturnGenerator` and implementing it only with transparent methods into inner type looks like a lot of "noise", of unnecessary code I think.

Would you suggest a better way to preserve the previous pattern while having a transparent struct?

Same for Dart generator

Reply:
Hmm, are we sure SyncReturn will always *purely* delegate, and never develop its own logic? If not, it may be reasonable to do the delegating (though it is verbose).

As for verbose code for delegate: https://crates.io/crates/delegate maybe we can use a crate

Reply:
So I made use of delegate, restoring the existing pattern. Howver couldn't exactly replicate the exact same pattern in the enum pattern: `SyncReturn` is transparent which requires it to instantiate the inner generator at some point.

I considered involving lazy field like a `RefCell` but even this one involves defining it for the `ty.rs`s. Doing it in the `new` makes sense to me, and we could use a new `new` constructor for all other types, but I'm limiting the impact of the changes for now.

What do you think?